pypit/arms.py

```diff
@@ -53,10 +66,32 @@ def ARMS(spectrograph, fitstbl, setup_dict):
     basenames = [None]*numsci  # For fluxing at the very end
 
     # Spectrometer class
-    spectro_class = spectro_utils.load_spec_class(spectrograph=spectrograph)
+    if spectrograph is None:
+        # Set spectrograph from FITS table instrument header
+        # keyword.
+        if par is not None and par['rdx']['spectrograph'] != fitstbl['instrume'][0]:
+            msgs.error('Specified spectrograph does not match instrument in the fits table!')
+        _spectrograph = load_spectrograph(spectrograph=fitstbl['instrume'][0])
+    elif isinstance(spectrograph, basestring):
+        _spectrograph = load_spectrograph(spectrograph=spectrograph)
+    elif isinstance(spectrograph, Spectrograph):
+        _spectrograph = spectrograph
+    else:
+        raise TypeError('Could not instantiate Spectrograph!')
```

Comment:
Is there a  reason why a TypeError is raised, and not a msgs.error?

Reply:
I've been assigned #329, so I'll aim to deal with this in a future PR.  To be honest, I actually fine the Exception types useful as a quick assessment of what's going on.  I don't find `PypitError` as useful, but we should talk about this in a future PR.  Punting on this for now.